modules/ingest-common/src/test/java/org/elasticsearch/ingest/common/DotExpanderProcessorTests.java

```diff
@@ -147,8 +139,8 @@ public void testEscapeFields_doNothingIfFieldNotInSourceDoc() throws Exception {
         // abc.def does not exist in source, so don't mutate document
         DotExpanderProcessor processor = new DotExpanderProcessor("_tag", null, null, "abc.def");
         processor.execute(document);
-        // hasField returns false since it requires the expanded form, which is not expanded since we did not ask for it to be
-        assertFalse(document.hasField("foo.bar"));
```

Comment:
It now can resolve the dotted field name `['foo.bar']`, however `['foo']` does not exist as a standalone object.